test/RachisTests/SubscriptionFailoverWIthWaitingChains.cs

```diff
@@ -222,7 +222,11 @@ public async Task SubscriptionShouldReconnectOnExceptionInTcpListener()
                 subsWorker.OnSubscriptionConnectionRetry += ex =>
                 {
                     Assert.NotNull(ex);
-                    Assert.True(ex.GetType() == typeof(IOException) || ex.GetType() == typeof(EndOfStreamException));
+                    
+                    if (ex is AggregateException ae)
```

Comment:
why those fixes were made? Asking because maybe we need to migrate them to 4.2 ?

Reply:
It's because ConnectSecuredSocket() (used to be connectAsync + WrapSsl + Negotiate) now returns an AggregateException instead of a single error. And because other places in this code rely on these errors I had to unwrap the AE and check each of them (otherwise tests fail)